File: examples/crud-http-server/src/main/java/com/tuhuynh/crud/router/CatRouter.java
package com.tuhuynh.crud.router;

import com.tuhuynh.crud.handlers.CatHandler;
import com.tuhuynh.crud.storage.MongoDB;
import com.tuhuynh.jerrymouse.core.bio.HttpRouter;
import lombok.val;

public final class CatRouter {
    public static HttpRouter getRouter() {
        val mongoClient = MongoDB.init();
        val catHandler = new CatHandler(mongoClient);
        val router = new HttpRouter();
        router.get("/", catHandler::getCats);
        router.post("/", catHandler::addCat);
        return router;
    }
}

File: examples/scala-crud-http-server/src/main/scala/com/tuhuynh/scala/crud/routers/CatRouter.scala
package com.tuhuynh.scala.crud.routers

import com.tuhuynh.jerrymouse.core.bio.HttpRouter
import com.tuhuynh.scala.crud.handlers.CatHandler

object CatRouter {
  def getRouter: HttpRouter = {
    val router = new HttpRouter()
    router.get("/", CatHandler.getCats)
    router
  }
}
